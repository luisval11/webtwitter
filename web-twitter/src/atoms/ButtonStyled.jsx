import React from "react";
import PropTypes from "prop-types";
import { Button } from "react-bootstrap";

export const ButtonStyled = ({ text, ...rest }) => {
	return <Button {...rest}>{text}</Button>;
};

ButtonStyled.propTypes = {
	text: PropTypes.string,
	rest: PropTypes.object,
};

export default ButtonStyled;
