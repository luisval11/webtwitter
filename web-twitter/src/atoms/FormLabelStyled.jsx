import React from "react";
import PropTypes from "prop-types";
import { Form } from "react-bootstrap";

export const FormLabelStyled = ({ text, htmlFor, ...rest }) => {
	return (
		<Form.Label htmlFor={htmlFor} {...rest}>
			{text}
		</Form.Label>
	);
};

FormLabelStyled.propTypes = {
	text: PropTypes.string,
	htmlFor: PropTypes.string,
	rest: PropTypes.object,
};

export default FormLabelStyled;
