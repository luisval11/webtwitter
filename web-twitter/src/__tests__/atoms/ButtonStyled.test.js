import { mount } from "enzyme";
import { ButtonStyled } from "../../atoms/ButtonStyled";

describe("Test on Button", () => {
	test("should match initial snapshot", () => {
		let wrapper = mount(<ButtonStyled text="InitialTest" />);

		expect(wrapper).toMatchSnapshot();
	});

	test("should have the text 'Button'", () => {
		const text = "Button";
		let wrapper = mount(<ButtonStyled text={text} />);
        
		expect(wrapper.find("button").text().trim()).toBe(text);
	});
});
