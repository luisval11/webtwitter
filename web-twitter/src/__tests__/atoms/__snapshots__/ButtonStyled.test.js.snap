// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Button should match initial snapshot 1`] = `
<button
  className="btn btn-primary"
  disabled={false}
  type="button"
>
  InitialTest
</button>
`;
